token/filetoken/filetoken.go

```diff
@@ -0,0 +1,119 @@
+package filetoken
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/puppetlabs/pe-sdk-go/log"
+	"github.com/puppetlabs/pe-sdk-go/token"
+)
+
+// fileToken struct
+type fileToken struct {
+	path string
+}
+
+// NewFileToken constructs a new filetoken
+func NewFileToken(path string) token.Token {
+	fileToken := fileToken{path: path}
+	return &fileToken
+}
+
+// Read reads the token from a file and returns a valid token.
+// If the token is not valid an error will be return
+func (ft *fileToken) ReadTokenFile() (string, error) {
```

Comment:
I think method name `Read` should be better, since it is already part of `fileToken`